src/pytorch_lightning/overrides/distributed.py

```diff
@@ -109,60 +109,6 @@ def __iter__(self) -> Iterator[List[int]]:
         return iter(indices)
 
 
-class _DatasetSamplerWrapper(Dataset):
-    """Dataset to create indexes from `Sampler` or `Iterable`"""
-
-    def __init__(self, sampler: Union[Sampler, Iterable]) -> None:
-        if not isinstance(sampler, Sized):
-            raise MisconfigurationException(
-                "You seem to have configured a sampler in your DataLoader which"
-                " does not provide `__len__` method. The sampler was about to be"
-                " replaced by `DistributedSamplerWrapper` since `replace_sampler_ddp`"
-                " is True and you are using distributed training. Either provide `__len__`"
-                " method in your sampler, remove it from DataLoader or set `replace_sampler_ddp=False`"
-                " if you want to handle distributed sampling yourself."
-            )
-        if len(sampler) == float("inf"):
-            raise MisconfigurationException(
-                "You seem to have configured a sampler in your DataLoader which"
-                " does not provide finite `__len__` method. The sampler was about to be"
-                " replaced by `DistributedSamplerWrapper` since `replace_sampler_ddp`"
-                " is True and you are using distributed training. Either provide `__len__`"
-                " method in your sampler which returns a finite number, remove it from DataLoader"
-                " or set `replace_sampler_ddp=False` if you want to handle distributed sampling yourself."
-            )
-        self._sampler = sampler
-        # defer materializing an iterator until it is necessary
-        self._sampler_list: Optional[List[Any]] = None
-
-    def __getitem__(self, index: int) -> Any:
-        if self._sampler_list is None:
-            self._sampler_list = list(self._sampler)
-        return self._sampler_list[index]
-
-    def __len__(self) -> int:
-        return len(self._sampler)
-
-    def reset(self) -> None:
-        """Reset the sampler list in order to get new sampling."""
-        self._sampler_list = list(self._sampler)
-
-
-class DistributedSamplerWrapper(DistributedSampler):
```

Comment:
Moved these classes without change (only change was neutralizing the mention of PyTorch Lightning in the docstring)